samples/gatsby-preview/src/index.jsx

```diff
@@ -0,0 +1,165 @@
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+
+    let parameters = this.props.sdk.parameters
+    let projectId = parameters.installation.projectId
+    let contentTypeSlug = parameters.instance.contentTypeSlug
+    let isAutoUpdate = parameters.instance.isAutoUpdate
+
+    this.state = {
+      projectId: projectId,
+      projectUrl:
+        "https://" + projectId + "-preview.gtsb.io/" + contentTypeSlug + "/",
+      isAutoUpdate: isAutoUpdate,
+      webhookUrl:
+        "https://backends.ctffns.net/gatsby-preview-proxy/" + projectId,
+      debounceInterval: null
+    }
+  }
+
+  componentDidMount = () => {
+    this.detachFns = []
+
+    this.detachFns.push(this.props.sdk.entry.onSysChanged(this.onSysChanged))
+
+    this.props.sdk.window.startAutoResizer()
+  }
+
+  componentWillUnmount = () => {
+    this.detachFns.forEach(detach => detach())
+  }
+
+  constructState = () => {
+    return
+  }
+
+  onError = error => {
+    this.setState({ working: false })
+    this.props.sdk.notifier.error(error.message)
+  }
+
+  onSysChanged = () => {
+    if (!this.state.isAutoUpdate) {
+      return
+    }
+    if (this.state.debounceInterval) {
+      clearInterval(this.state.debounceInterval)
+    }
+    this.state.debounceInterval = setInterval(this.refreshGatsbyPreview, 1000)
+  }
+
+  refreshGatsbyPreview = () => {
+    clearInterval(this.state.debounceInterval)
```

Comment:
How about the approach with `this.updated`? You setting and clearing intervals all the time is hard to follow.